Source/utils/display.cpp

```diff
@@ -268,6 +269,27 @@ bool SpawnWindow(const char *lpWindowName)
 	return ghMainWnd != nullptr;
 }
 
+#ifndef USE_SDL1
+void ReinitializeTexture()
+{
+	if (texture)
+		texture.reset();
+
+	auto quality = fmt::format("{}", static_cast<int>(*sgOptions.Graphics.scaleQuality));
+	SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, quality.c_str());
+
+	texture = SDLWrap::CreateTexture(renderer, SDL_PIXELFORMAT_RGB888, SDL_TEXTUREACCESS_STREAMING, gnScreenWidth, gnScreenHeight);
+}
+
+void ReinitializeIntegerScale()
+{
+	if (SDL_RenderSetIntegerScale(renderer, *sgOptions.Graphics.integerScaling ? SDL_TRUE : SDL_FALSE) < 0) {
+		ErrSdl();
+	}
+	dx_resize();

```

Comment:
The latest commit merely updates the window size without updating the render logical size. Since the logical size doesn't change, there is no need to reinitialize the texture or the global back buffer.